rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/SelectionManager.java

```diff
@@ -428,22 +430,26 @@ private static boolean electedToRunOnAllNodes(Criteria criteria) {
     /**
      * Checks is all scripts are authorized. If not throws an exception.
      */
-    private void checkAuthorizedScripts(List<SelectionScript> scripts) {
+    private boolean checkAuthorizedScripts(List<SelectionScript> scripts) {
         updateAuthorizedScriptsSignatures();
         if (authorizedSelectionScripts == null || scripts == null)
-            return;
+            return true;
+
+        boolean allAuthorized = true;
```

Comment:
"boolean allAuthorized = true;" should be before "return true;"

Reply:
Moving the statement will add an useless allocation before finishing the method. Why are you pushing to do that @nebilbenmabrouk ?

Reply:
@MaelAudren because no instructions will be executed after the return statement

Reply:
@MaelAudren sorry I thought that both instructions are in the same block (because of the missing brackets).